src/components/apps/PublishAppDialog.js

```diff
@@ -0,0 +1,409 @@
+                                required={true}
+                                component={FormMultilineTextField}
+                                validate={(value) =>
+                                    nonEmptyField(value, i18nUtil)
+                                }
+                                helperText={t("inputDescHelpText")}
+                            />
+                            <Field
+                                name="parameterDesc"
+                                label={t("parameterDescLabel")}
+                                id={build(parentId, ids.PUBLISH.PARAMS)}
+                                required={true}
+                                component={FormMultilineTextField}
+                                validate={(value) =>
+                                    nonEmptyField(value, i18nUtil)
+                                }
+                            />
+                            <Field
+                                name="outputDesc"
+                                label={t("outputDescLabel")}
+                                id={build(parentId, ids.PUBLISH.OUTPUT)}
+                                required={true}
+                                component={FormMultilineTextField}
+                                validate={(value) =>
+                                    nonEmptyField(value, i18nUtil)
+                                }
+                                helperText={t("outputDescHelpText")}
+                            />
+                            <Field
+                                name="licenseType"
+                                label={t("licenseType")}
+                                id={build(parentId, ids.PUBLISH.LICENSE_TYPE)}
+                                component={FormTextField}
+                                helperText={t("licenseTypeHelp")}
+                            />
+                            <Field
+                                name="licenseLink"
+                                label={t("licenseLink")}
+                                id={build(parentId, ids.PUBLISH.LICENSE_LINK)}
+                                component={FormTextField}
+                                validate={(value) =>
+                                    urlField(value, i18nUtil, false)
+                                }
+                            />
+
+                            <Typography>{t("attributionLinks")}</Typography>
+                            <FieldArray
+                                name="references"
+                                render={(arrayHelpers) => (
+                                    <Grid container spacing={3}>
+                                        {values.references &&
+                                        values.references.length > 0 ? (
+                                            values.references.map(
+                                                (link, index) => (
+                                                    <>
```

Comment:
This fragment can be removed - should help the components from looking so squished in the file

Reply:
I don't think this one can be removed, since this `map` is returning multiple `Grid` items.

Reply:
✅ . May be I can extract this into its own component.

Reply:
Ah, you're right. I think because it's squished, I thought it was all under one grid. I need vertical helper lines like in my IDE 😄